node/src/Commands.ts

```diff
@@ -1974,6 +1974,61 @@ export function createXTrim(
     return createCommand(RequestType.XTrim, args);
 }
 
+export enum StreamIdBoundary {
+    /**
+     * Stream ID boundary used to specify the maximum stream entry ID. Can be used in the `XRANGE` or `XREVRANGE` commands
+     * to get the last stream ID.
+     */
+    MaxId = "+",
+    /**
+     * Stream ID boundary used to specify the minimum stream entry ID. Can be used in the `XRANGE` or `XREVRANGE` commands
+     * to get the first stream ID.
+     */
+    MinId = "-",
+}
+
+export type StreamRangeBound =
+    | StreamIdBoundary 
+    /**
+     * Stream ID boundary used to specify a range of IDs to search. Stream ID bounds can be complete with
+     * a timestamp and sequence number separated by a dash ("-"), for example "1526985054069-0". Stream ID bounds can also
+     * be incomplete, with just a timestamp. Can be specified as inclusive or exclusive, where inclusive is the default.
+     */
+    | {
+          isInclusive?: boolean;
+          id: string | number;
+      };
+
+function addRangeBound(rangeBound: StreamRangeBound): string {
+    if (rangeBound === StreamIdBoundary.MinId || rangeBound === StreamIdBoundary.MaxId) {
+        return rangeBound;
+    }
+
+    if (rangeBound.isInclusive == false) {
+        return "(" + rangeBound.id.toString();
+    }
+    return rangeBound.id.toString();
+}
+
+/**
+ * @internal
+ */
+export function createXRange(
+    key: string,
+    start: StreamRangeBound,
+    end: StreamRangeBound,
```

Comment:
You can reuse `ScoreBoundary` and delete code above
```suggestion
    start: ScoreBoundary<string>,
    end: ScoreBoundary<string>,
```

Reply:
What about timestamps?  Those are numbers.  I guess I could use `ScoreBoundary<string | number>`, but still, the docs for ScoreBoundary are all talking about sorted sets, which is confusing for users.  It would probably be better to have a generic Boundary that can be type aliased for ScoreBoundary and StreamBoundary.

Reply:
Timestamps are actually in format `<ts>-<n>`, where `n` is zero based index of message recevied in this ms of `ts`. For example `1723489748-0` - so it is not a pure number.
You can see how I used in #2085 for `XPENDING`

Reply:
Oh I see. I still don't know about using ScoreBoundary, but I see we're already using it for XPENDING anyway, so I guess I'll just go with it for now.  If it becomes a problem, we can always change it later.